rewrite-core/src/main/java/org/openrewrite/marker/RecipesThatMadeChanges.java

```diff
@@ -32,4 +34,70 @@ public class RecipesThatMadeChanges implements Marker {
     UUID id;
 
     Collection<Stack<Recipe>> recipes;
+
+    /**
+     * A list of recipe names and their associated options which have made changes to the AST
+     * @param indent string prepended to sub recipes for logging a hierarchical view
+     */
+    @Incubating(since = "7.22.0")
+    public List<String> recipeNamesAndOptionsThatMadeChanges(String indent) {
```

Comment:
I think I would probably have this method transform the stacks into recipe descriptors and return those. 

It can then be up to the caller to figure out how best to render them: 

So something like

```
/**
 * Return a list of recipes that have made changes as a hierarchy of descriptors. The method transforms the flat, stack-based representation into descriptors where children are grouped under their common parents.
 *
 * <PRE>
 *
 * List of changes: 
 * 
 * A->B->C
 * A->B->D
 * E->F
 * E->G
 * H
 *
 * Converted to a List of the following descriptors:
 *
 * A
 * |-B
 * | |-C
 * | |-D
 *
 * E
 * |-F
 * |-G
 * 
 *
 * H
 *
 * </PRE>
 */
public List<RecipeDescriptor> recipeNamesAndOptionsThatMadeChanges() {
```

Reply:
Thats a good suggestion, thanks Tyler